Tests/Acceptance/CheckoutTestCase.php

```diff
@@ -0,0 +1,149 @@
+<?php
+/**
+ * Shop System Plugins:
+ * - Terms of Use can be found under:
+ * https://github.com/wirecard/oxid-ee/blob/master/_TERMS_OF_USE
+ * - License can be found under:
+ * https://github.com/wirecard/oxid-ee/blob/master/LICENSE
+ */
+
+namespace Wirecard\Oxid\Tests\Acceptance;
+
+use Wirecard\Oxid\Model\Transaction;
+
+/**
+ * Acceptance test class for testing checkout flows.
+ */
+abstract class CheckoutTestCase extends BaseAcceptanceTestCase
+{
+    /**
+     * @var Wirecard\Oxid\Model\PaymentMethod
+     */
+    protected $paymentMethod;
+
+    /**
+     * @inheritdoc
+     */
+    protected function setUp()
+    {
+        parent::setUp();
+
+        $this->paymentMethod = $this->getPaymentMethod();
+
+        $this->activatePaymentMethod();
+        $this->addMockUser();
+        $this->addMockArticle();
+    }
+
+    /**
+     * Payment method getter.
+     */
+    abstract public function getPaymentMethod();
+
+    /**
+     * Activates the payment method.
+     */
+    public function activatePaymentMethod()
+    {
+        $this->executeSql("UPDATE `oxpayments` SET `OXACTIVE` = '1'
+            WHERE `OXID` = '{$this->paymentMethod::getName(true)}'");
+    }
+
+    /**
+     * Sets a payment method's payment action to "purchase".
+     */
+    public function setPaymentActionPurchase()
+    {
+        $this->executeSql("UPDATE `oxpayments` SET `WDOXIDEE_TRANSACTIONACTION` = '" . Transaction::ACTION_PAY .
+            "' WHERE `OXID` = '{$this->paymentMethod::getName(true)}'");
+    }
+
+    /**
+     * Sets a payment method's payment action to "authorize".
+     */
+    public function setPaymentActionAuthorize()
+    {
+        $this->executeSql("UPDATE `oxpayments` SET `WDOXIDEE_TRANSACTIONACTION` = '" . Transaction::ACTION_RESERVE .
+            "' WHERE `OXID` = '{$this->paymentMethod::getName(true)}'");
+    }
+
+    /**
+     * Adds a demo user to the database.
+     */
+    public function addMockUser()
+    {
+        $this->executeSql("INSERT INTO `oxuser`
```

Comment:
The test data shall be in a separate file(json), for example we have it like this in [woocommerce](https://github.com/wirecard/woocommerce-ee/blob/master/wirecard-woocommerce-extension/tests/_data/CustomerData.json). In the future we plan on randomizing the test data.